tests/reference/asr-test_dict2-4587f02.stderr

```diff
@@ -0,0 +1,5 @@
+semantic error: Type mismatch in dictionary key, the types must be compatible
+ --> tests/errors/test_dict2.py:4:5
+  |
+4 |     y[1] = -3
+  |     ^^^^^^^^^ type mismatch ('i32' and 'str')
```

Comment:
This can be an improvement in the error message:
```diff
diff --git a/tests/reference/asr-test_dict2-4587f02.stderr b/tests/reference/asr-test_dict2-4587f02.stderr
index 087b64b31..5cb006aeb 100644
--- a/tests/reference/asr-test_dict2-4587f02.stderr
+++ b/tests/reference/asr-test_dict2-4587f02.stderr
@@ -1,5 +1,5 @@
 semantic error: Type mismatch in dictionary key, the types must be compatible
- --> tests/errors/test_dict2.py:4:5
+ --> tests/errors/test_dict2.py:4:7
   |
 4 |     y[1] = -3
-  |     ^^^^^^^^^ type mismatch ('i32' and 'str')
+  |       ^ type mismatch (found 'i32' -- expected 'str')
```
```diff
diff --git a/src/lpython/semantics/python_ast_to_asr.cpp b/src/lpython/semantics/python_ast_to_asr.cpp
index b94959efa..d62462787 100644
--- a/src/lpython/semantics/python_ast_to_asr.cpp
+++ b/src/lpython/semantics/python_ast_to_asr.cpp
@@ -1833,8 +1833,8 @@ public:
                             diag.add(diag::Diagnostic(
                                 "Type mismatch in dictionary key, the types must be compatible",
                                 diag::Level::Error, diag::Stage::Semantic, {
-                                    diag::Label("type mismatch ('" + ktype + "' and '" + totype + "')",
-                                            {x.base.base.loc})
+                                    diag::Label("type mismatch (found '" + ktype + "' -- expected '" + totype + "')",
+                                            {key->base.loc})
                                 })
                             );
                             throw SemanticAbort();
@@ -1845,8 +1845,8 @@ public:
                             diag.add(diag::Diagnostic(
                                 "Type mismatch in dictionary value, the types must be compatible",
                                 diag::Level::Error, diag::Stage::Semantic, {
-                                    diag::Label("type mismatch ('" + vtype + "' and '" + totype + "')",
-                                            {x.base.base.loc})
+                                    diag::Label("type mismatch (found '" + vtype + "' -- expected '" + totype + "')",
+                                            {val->base.loc})
                                 })
                             );
                             throw SemanticAbort();
```

Reply:
Maybe use a comma instead of `--`.

Reply:
Fixed. Thanks!